src/main/java/com/sparta/chairingproject/domain/reservation/service/ReservationService.java

```diff
@@ -21,21 +25,48 @@ public class ReservationService {
 
     private final ReservationRepository reservationRepository;
     private final StoreRepository storeRepository;
+    private final MemberRepository memberRepository;
 
+    /* 일반 사용자 */
 
     public ReservationResponse createReservation(Long storeId, CreateReservationRequest req, UserDetailsImpl authUser) {
+        Long memberId = resolveMember(req, authUser).getId();
+
         Store store = storeRepository.findById(storeId).orElseThrow(
                 () -> new GlobalException(ExceptionCode.NOT_FOUND_STORE)
         );
 
-        Long memberId = (req.getMemberId() == null || req.getMemberId() == 0) ? authUser.getMember().getId() : req.getMemberId();
-
         Reservation savedReservation = reservationRepository.save(req.toEntity(memberId, store));
 
         return savedReservation.toResponse();
     }
 
+    public ReservationResponse cancelReservation(Long reservationId, RequestDto req, UserDetailsImpl authUser) {
+        Long memberId = resolveMember(req, authUser).getId();
+
+        Reservation reservation = reservationRepository.findById(reservationId).orElseThrow(
+                () -> new GlobalException(ExceptionCode.RESERVATION_NOT_FOUND)
```

Comment:
I recommend to use this import sentence
`import static com.sparta.chairingproject.config.exception.enums.ExceptionCode.*;`